### PR TITLE
[Feature][Transform-V2] Support SqlTransform Not Like Expression

### PR DESCRIPTION
**Pull request by zhilinli123**
preTask:https://github.com/apache/seatunnel/pull/5456
### How was this patch tested?
Add e2e conf test data

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFilter.java [hunk 1]
```diff
@@ -66,7 +66,15 @@ public boolean executeFilter(Expression whereExpr, Object[] inputFields) {
...
-            return likeExpr((LikeExpression) whereExpr, inputFields);
+            boolean isNotLike = ((LikeExpression) whereExpr).isNot();
+            // like SQL parsing
+            if (!isNotLike) {
+                return likeExpr((LikeExpression) whereExpr, inputFields);
+            }
+            // not like SQL parsing
+            if (isNotLike) {
+                return notLikeExpr((LikeExpression) whereExpr, inputFields);
+            }
```
Comment (Hisoka-X) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFilter.java]:
```suggestion
            boolean isNotLike = ((LikeExpression) whereExpr).isNot();
            if (isNotLike) {
                return notLikeExpr((LikeExpression) whereExpr, inputFields);
            } else {
                return likeExpr((LikeExpression) whereExpr, inputFields);
            }
```

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFilter.java [hunk 2]
```diff
@@ -148,7 +156,7 @@ private boolean inExpr(InExpression inExpression, Object[] inputFields) {
     }
 
     /**
-     * Like expression filter, unsupported yet
+     * like syntax expression filters
```
Comment (Hisoka-X) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFilter.java]:
```suggestion
     * Like expression filter
```

**Review by Hisoka-X — COMMENTED** [review]
I think the test case not enough. We need to cover all `like` and `not like` situations. Including prefixes, suffixes, etc, all situations contained in the code.

**Comment by zhilinli123** [comment]
image
image

**Comment by zhilinli123** [comment]
PTAL: @Hisoka-X @EricJoy2048 @hailin0

**Review by EricJoy2048 — DISMISSED** [review]

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFilter.java [hunk 3]
```diff
@@ -162,11 +170,29 @@ private boolean likeExpr(LikeExpression likeExpression, Object[] inputFields) {
...
+        }
+        if (regex.length() >= 3 && regex.substring(regex.length() - 3).endsWith("_.*")) {
+            regex = regex.substring(0, regex.length() - 3) + "..*";
+        }
+        if (regex.startsWith("'") && regex.endsWith("'")) {
+            regex = regex.substring(0, regex.length() - 1).substring(1);
+        }
         if (regex.startsWith("'") && regex.endsWith("'")) {
             regex = regex.substring(0, regex.length() - 1).substring(1);
         }
```
Comment (Hisoka-X) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFilter.java]: Why this logic exist twice?

Comment (zhilinli123) [reply seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFilter.java]: sorry,My question, thank you for your review

**Review by EricJoy2048 — APPROVED** [review]

**Comment by zhilinli123** [comment]
thanks. review  PTAL: @Hisoka-X

**Review by Hisoka-X — APPROVED** [review]
Thanks @zhilinli123 !

**Pull request merged**